src/snowflake/snowpark/_internal/analyzer/datatype_mapper.py

```diff
@@ -51,6 +52,31 @@ def str_to_sql(value: str) -> str:
     return f"'{sql_str}'"
 
 
+def str_to_sql_for_year_month_interval(
+    value: str, datatype: YearMonthIntervalType
+) -> str:
+    # Extract interval value from strings like "INTERVAL 1-7 YEAR TO MONTH"
+    # Examples: "1-7" -> INTERVAL '1-7' YEAR TO MONTH, "1" -> INTERVAL '1' YEAR
+    # "1-7" -> INTERVAL '7' MONTH
```

Comment:
how is "1-7" mapped to both `INTERVAL '1-7' YEAR TO MONTH` and `INTERVAL '7' MONTH`